src/rust/catnap/futures/accept.rs

```diff
@@ -62,46 +66,63 @@ impl AcceptFuture {
     }
 }
 
-//==============================================================================
+//======================================================================================================================
 // Trait Implementations
-//==============================================================================
+//======================================================================================================================
 
 /// Future Trait Implementation for Accept Operation Descriptors
 impl Future for AcceptFuture {
     type Output = Result<RawFd, Fail>;
 
     /// Polls the target [AcceptFuture].
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
-        let self_: &AcceptFuture = self.get_mut();
-        match socket::accept(self_.fd as i32) {
+        let self_: &mut AcceptFuture = self.get_mut();
+        match unsafe {
+            let mut address_len: libc::socklen_t = 0;
```

Comment:
This one too.

Reply:
You are correct. I have fixed this. Thanks for checking this.